sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -444,11 +444,11 @@ case class DayOfWeek(child: Expression) extends UnaryExpression with ImplicitCas
       val cal = classOf[Calendar].getName
       val c = ctx.freshName("cal")
       val dtu = DateTimeUtils.getClass.getName.stripSuffix("$")
-      ctx.addMutableState(cal, c, s"""$c = $cal.getInstance($dtu.getTimeZone("UTC"));""")
```

Comment:
this avoids calling `getInstance` every time.

Reply:
but this is done every time, since we are crating a new one for each expression

Reply:
yea it's done for each expression, but each expression is evaluated many times.

Reply:
now I got it, sorry. I think the same thing can be said also for the timezone. Thus I am closing this. Sorry again.